panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -1094,6 +1081,36 @@ def get_certificates(self) -> dict:
 
         return result
 
+    def get_update_schedules(self) -> dict:
+        """Get schedules for all dynamic updates.
+
+        This method gets scheduled dynamic updates on a device. This includes the ones pushed from Panorama.
+
+        The actual API command is `<show><config><effective-running><xpath>devices/entry/deviceconfig/system/update-schedule</xpath></effective-running></config></show>`.
+
+        # Returns
+
+        dict: All dynamic updates schedules, key is the entity type to update, like: threats, wildfire, etc.
+
+        ```python showLineNumbers title="Sample output"
+        {'threats': {'recurring': {'weekly': {'action': 'download-only',
+                                      'at': '01:02',
+                                      'day-of-week': 'wednesday'}}},
+        'wildfire': {'recurring': {'real-time': None}}}
+        ```
+
+        """
+        schedules = self.op_parser(
+            cmd="<show><config><effective-running><xpath>devices/entry/deviceconfig/system/update-schedule</xpath></effective-running></config></show>",
+            cmd_in_xml=True,
+        )
+        if schedules is None:
+            return {}
+        if "update-schedule" not in schedules:
+            return {}
```

Comment:
We can put this into a single if condition; as the "or" condition will be evaluated if "schedules" is not None.

```suggestion
        if schedules is None or "update-schedule" not in schedules:
            return {}
```